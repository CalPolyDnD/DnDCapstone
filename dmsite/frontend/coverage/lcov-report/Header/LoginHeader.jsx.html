<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Header/LoginHeader.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\prettify.css" />
    <link rel="stylesheet" href="..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\index.html">All files</a> / <a href="index.html">Header</a> LoginHeader.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/2</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import Logo from '../images/DnDLogo2.png';
&nbsp;
const LoginHeader = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >  return (</span>
    &lt;div style={{ height: '64px', backgroundColor: '#000000' }} id="headerContainer"&gt;
      &lt;a className="d-inline-flex px-2 justify-self-center mx-auto" href="/" id="logoBanner"&gt;
        &lt;img id="logo" alt="" width="90" height="70" src={Logo} /&gt;
        &lt;h1
          className="m-0"
          style={{
            color: 'white',
            textDecoration: 'none',
            height: '50px',
            alignSelf: 'center',
            lineHeight: '50px',
          }}
        &gt;
          DataMaster
        &lt;/h1&gt;
      &lt;/a&gt;
    &lt;/div&gt;
  );
};
&nbsp;
export default LoginHeader;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Mar 05 2019 11:02:06 GMT-0800 (Pacific Standard Time)
</div>
</div>
<script src="..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\sorter.js"></script>
</body>
</html>
