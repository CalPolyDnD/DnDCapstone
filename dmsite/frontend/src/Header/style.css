#logo {
  align-self: center;
  justify-content: center;
}

#header {
  align-items: center;
  justify-content: center;
}

#searchForm {
  flex-direction: row;
  align-self: center;
  justify-content: center;
}

#searchBar {
  flex: auto;
  align-self: center;
  justify-content: center;
}

#header-container {
  justify-content: center;
  align-self: center;
}

#logoBanner {
  height: 64px;
}

.navbar {
  height: 65px;
}

.navbar .navbar-brand {
  line-height: 32px;
}

#buttonContainer {
  align-items: center;
  height: inherit;
  margin-left: auto;
  margin-right: 8px;
  justify-self: flex-end;
}

#headerContainer {
  display: inline-flex;
  flex: auto;
  flex-direction: row;
  width: 100%;
}

.ant-btn {
  line-height: 50%;
}

h2, h1 {
  margin: 0;
}
