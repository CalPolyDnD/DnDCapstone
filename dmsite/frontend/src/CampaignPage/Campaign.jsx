import React from 'react';
import {
  ListGroup, ListGroupItem, Card, CardBody, Button, CardHeader,
} from 'reactstrap';
import './style.css';
import axios from 'axios';

const FETCH_CURRENT_USER_URL = 'http://localhost:8000/rest-auth/user/';
const FETCH_CAMPAIGNS_URL = 'http://localhost:8000/get_campaigns';

class Campaign extends React.Component {
  constructor() {
    super();
    this.state = {
      campaigns: [],
    };
  }

  onClick() {
    this.props.history.push(`/a/${this.props.a}`);
    window.location.reload();


  }

  componentDidMount() {
    axios.get(FETCH_CURRENT_USER_URL)
      .then((userRes) => {
        axios.post(FETCH_CAMPAIGNS_URL, {
          user: userRes.data.email,
        })
          .then((campRes) => {
            this.setState({ campaigns: campRes.data });
          });
      });
  }

  getCampaignList() {
    const jsonObj = [];
    const { campaigns } = this.state;
    const { history } = this.props;
    for (let count = 0; count < campaigns.length; count += 1) {
      const result = campaigns[count];
      jsonObj.push(
        <ListGroupItem
          onClick={() => { history.push(`/home${campaigns.name}`); }}
          style={{ backgroundColor: '#3d3d3d', color: 'white' }}
        >
          {result.name}
        </ListGroupItem>,
      );
    }
    return jsonObj;
  }

  render() {
    return (
      <div id="div" style={{ paddingTop: '5%' }}>
        <Card id="card" style={{ borderWidth: 0 }}>
          <CardHeader tag="h3" align="center" style={{ backgroundColor: '#303030', color: 'white' }}>Campaigns</CardHeader>
          <CardBody style={{ backgroundColor: '#3d3d3d', borderWidth: 0 }}>
            <ListGroup align="center" className="campaign-list" flush>
              {this.getCampaignList()}
            </ListGroup>
            <Button color="primary" size="md" className="btn-block mt-3">Create New Campaign</Button>
            <Button color="secondary" size="md" className="btn-block mt-3"
                    disabled={this.state.campaigns.length === 0}
                    onClick={this.onClick}>Save & Continue </Button>
          </CardBody>
        </Card>
      </div>
    );
  }
}

export default Campaign;
