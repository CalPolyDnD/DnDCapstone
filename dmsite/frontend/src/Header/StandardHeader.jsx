import React from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import ReactRouterPropTypes from 'react-router-prop-types';
import { connect } from 'react-redux';
import {
  Menu,
  Button,
  Dropdown,
  Input,
  Select,
} from 'antd';

import Logo from '../images/DnDLogo2.png';
import './style.css';
import * as actions from '../store/actions/auth';

const MenuItem = Menu.Item;
const { Search } = Input;
const InputGroup = Input.Group;
const { Option } = Select;

const SEARCH_FIELDS = {
  name: 'Name',
  classification: 'Classification',
  attribute: 'Attribute',
};

class StandardHeader extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      searchFilterOrder: [
        SEARCH_FIELDS.name,
        SEARCH_FIELDS.classification,
        SEARCH_FIELDS.attribute,
      ],
      searchbarFilter: 'Name',
    };
    this.switchFilter = this.switchFilter.bind(this);
  }

  switchFilter(index) {
    const { searchFilterOrder } = this.state;
    this.setState(prevState => (
      {
        ...prevState,
        searchbarFilter: searchFilterOrder[index],
      }));
  }

  renderSearchBar() {
    const { history } = this.props;
    const { searchFilterOrder, searchbarFilter } = this.state;
    return (
      <InputGroup
        compact
        className="align-self-center px-4"
      >
        <Select
          dropdownMatchSelectWidth={false}
          defaultValue={0}
          onSelect={this.switchFilter}
          className="float-left"
          style={{ backgroundColor:'#303030'}}
        >
          <Option value={0} style={{ backgroundColor:'#303030', color:'white'}}>{ searchFilterOrder[0] }</Option>
          <Option value={1} style={{ backgroundColor:'#303030', color:'white'}}>{ searchFilterOrder[1] }</Option>
          <Option value={2} style={{ backgroundColor:'#303030', color:'white'}}>{ searchFilterOrder[2] }</Option>
        </Select>
        <Search
          placeholder={`Search Campaign by ${searchbarFilter}`}
          onSearch={(searchbarText) => { history.push(`/search/${searchbarFilter}?=${searchbarText}`); }}
          enterButton
          style={{ width: '40%', height: '32px', background:'#303030' }}
        />
      </InputGroup>
    );
  }

  renderProfileDropdown() {
    const { isAuthenticated, logout } = this.props;
    return (
      <Menu>
        {
          isAuthenticated
            ? ([
              <MenuItem>
                <a href="/profile" key="profile">Profile</a>
              </MenuItem>,

              <MenuItem>
                <a onClick={logout} href="/login" key="logout" style={{ color: 'red' }}>Logout</a>
              </MenuItem>,
            ]
            )
            : (
              <MenuItem>
                <a href="/login/" key="login">Login</a>
              </MenuItem>

            )
        }
      </Menu>
    );
  }

  renderNavItems() {
    return (
      <div className="d-inline-flex px-2" id="buttonContainer">
        <Button
          title="Help"
          icon="question"
          shape="circle"
          ghost
          className="mx-1"
          htmlType="button"
        />
        <Button onClick={() => { this.props.history.push('/profile/'); }}
          title="Settings"
          icon="setting"
          shape="circle-outline"
          ghost
          className="mx-1"
          htmlType="button"
        />
        <Dropdown overlay={this.renderProfileDropdown()}>
          <Button
            icon="user"
            shape="circle-outline"
            ghost
            className="mx-1"
            htmlType="button"
          />
        </Dropdown>
      </div>
    );
  }

  render() {
    return (
      <div style={{ height: '64px', background: '#000000' }} id="headerContainer">
        <div style={{ background: '#000000' }} >
        <a className="d-inline-flex px-2" href="/home/" id="logoBanner">
          <img id="logo" alt="" width="65" height="50" src={Logo} />
          <h2
            style={{
              color: 'white',
              textDecoration: 'none',
              height: '50px',
              alignSelf: 'center',
              lineHeight: '50px',
            }}
          >
            DataMaster
          </h2>
        </a>
        </div>
        { this.renderSearchBar() }
        { this.renderNavItems() }
      </div>
    );
  }
}

StandardHeader.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  history: ReactRouterPropTypes.history.isRequired,
  logout: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  logout: () => dispatch(actions.logout()),
})


export default connect(mapDispatchToProps)(withRouter(StandardHeader));
