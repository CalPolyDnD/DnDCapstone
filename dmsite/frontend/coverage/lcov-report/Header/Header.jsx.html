<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Header/Header.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\prettify.css" />
    <link rel="stylesheet" href="..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\index.html">All files</a> / <a href="index.html">Header</a> Header.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import ReactRouterPropTypes from 'react-router-prop-types';
&nbsp;
import './style.css';
import { Layout } from 'antd';
import LoginHeader from './LoginHeader';
import StandardHeader from './StandardHeader';
&nbsp;
&nbsp;
class Header extends React.Component {
  render() {
    const { location } = this.props;
    const AntHeader = Layout.Header;
    return (
      &lt;AntHeader className="pl-0 pr-0" style={ {position: 'fixed', zIndex: 1, width: '100%', color: '#000000'} } &gt;
        {
          location.pathname === '/login/'
            ? <span class="branch-0 cbranch-no" title="branch not covered" >&lt;LoginHeader {...this.props} /&gt;</span>
            : &lt;StandardHeader {...this.props} /&gt;
        }
      &lt;/AntHeader&gt;
    );
  }
}
&nbsp;
Header.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  history: ReactRouterPropTypes.history.isRequired,
  location: ReactRouterPropTypes.location.isRequired,
};
&nbsp;
export default withRouter(Header);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Feb 21 2019 10:47:18 GMT-0800 (Pacific Standard Time)
</div>
</div>
<script src="..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\sorter.js"></script>
</body>
</html>
