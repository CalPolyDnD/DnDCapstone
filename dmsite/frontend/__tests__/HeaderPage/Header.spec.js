
import 'jsdom-global/register';
import React from 'react';
import { mount, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Header from '../../src/Header/Header';
import LoginHeader from '../../src/Header/LoginHeader';
import StandardHeader from '../../src/Header/StandardHeader';
import Adapter from 'enzyme-adapter-react-16';
import Campaign from "../CampaignPage/Campaign.spec";

configure({ adapter: new Adapter() })

describe('Layout-Footer', () => {

	it('Header with authentication should render the component without errors', () => {

		const wrapper = mount(
			<MemoryRouter>
				<Header isAuthenticated={true} />
			</MemoryRouter>,
		);
		const tree = renderer
			.create(<Campaign />)
			.toJSON();
		expect(toJson(wrapper.find(Header))).toMatchSnapshot();
	});
	it('Header without authentication should render the component without errors', () => {

		const wrapper = mount(
			<MemoryRouter>
				<Header isAuthenticated={false} />
			</MemoryRouter>,
		);
		expect(toJson(wrapper.find(Header))).toMatchSnapshot();
	});
	it('should render the component without errors', () => {
		const wrapper = mount(
			<MemoryRouter>
				<LoginHeader/>
			</MemoryRouter>,
		);
		expect(toJson(wrapper.find(LoginHeader))).toMatchSnapshot();
	});
	it('should render the component without errors', () => {
		const wrapper = mount(
			<MemoryRouter>
				<StandardHeader />
			</MemoryRouter>,
		);
		expect(toJson(wrapper.find(StandardHeader))).toMatchSnapshot();
	});
})



