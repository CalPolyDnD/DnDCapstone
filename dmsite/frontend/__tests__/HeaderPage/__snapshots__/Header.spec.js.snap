// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout-Footer should render the component without errors 1`] = `
<LoginHeader>
  <div
    id="headerContainer"
    style={
      Object {
        "height": "64px",
      }
    }
  >
    <a
      className="d-inline-flex px-2 justify-self-center mx-auto"
      href="/"
      id="logoBanner"
    >
      <img
        alt=""
        height="70"
        id="logo"
        src="test-file-stub"
        width="90"
      />
      <h1
        className="m-0"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "height": "50px",
            "lineHeight": "50px",
            "textDecoration": "none",
          }
        }
      >
        DataMaster
      </h1>
    </a>
  </div>
</LoginHeader>
`;
